laske_export/exporter.py

```diff
@@ -130,63 +136,72 @@ def export_invoices(self, invoices):
+                                invoice.id, invoice.credited_invoice.id
+                            )
                         )
-                    )
-                else:
-                    self.write_to_output(
-                        " Not sending invoice id {} because the credited invoice is unknown.".format(
-                            invoice.id
+                    else:
+                        self.write_to_output(
+                            " Not sending invoice id {} because the credited invoice is unknown.".format(
+                                invoice.id
+                            )
                         )
-                    )
 
-                continue
+                    continue
 
-            # Zero amount invoices are currently sent to Laske. Uncomment this if desired otherwise.
-            # if invoice.billed_amount == 0:
-            #     self.write_to_output(' Not sending invoice id {} because the billed amount is 0'.format(invoice.id))
-            #     continue
+                if not invoice.invoicing_date:
+                    invoice.invoicing_date = now.date()
+                    invoice.save()
 
-            if not invoice.invoicing_date:
-                invoice.invoicing_date = now.date()
-                invoice.save()
+                sales_order = SalesOrder()
+                set_constant_laske_values(sales_order)
 
-            sales_order = SalesOrder()
-            set_constant_laske_values(sales_order)
+                adapter = InvoiceSalesOrderAdapter(
+                    invoice=invoice,
+                    sales_order=sales_order,
+                    receivable_type_rent=receivable_type_rent,
+                    receivable_type_collateral=receivable_type_collateral,
+                )
+                adapter.set_values()
 
-            adapter = InvoiceSalesOrderAdapter(
-                invoice=invoice,
-                sales_order=sales_order,
-                receivable_type_rent=receivable_type_rent,
-                receivable_type_collateral=receivable_type_collateral,
-            )
-            adapter.set_values()
+                sales_order.validate()
 
-            # TODO: check sales_order validity
-            sales_orders.append(sales_order)
-            log_invoices.append(invoice)
+                sales_orders.append(sales_order)
 
-            invoice_count += 1
+                invoice_count += 1
 
-            self.write_to_output(
-                " Added invoice id {} as invoice number {}".format(
-                    invoice.id, invoice.number
+                self.write_to_output(
+                    " Added invoice id {} as invoice number {}".format(
+                        invoice.id, invoice.number
+                    )
                 )
-            )
+
+                invoice_log_item.status = LaskeExportLogInvoiceStatus.SENT
+            except ValidationError as err:
+                logger.warning(err, exc_info=True)
```

Comment:
I think in addition to logging with the logger, we should also `self.write_to_output` some semi-human-readable error. The reasoning behind this is that the `batchrun` system which actually calls the repeating management commands logs their `stdout` and `stderr` streams. The batchrun logs are viewable in the UI which I've heard the users sometimes look at.

Reply:
Good point. `self.write_to_output` has been added.